Bottom: 54b954e2878106918c9a8b22f23522915099fab7
Top:    87d49d35849a2413bd1740722b52c372d76e3835
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-22 04:21:25 +0100

Fix bug with polygon boolean operation PBO_XOR (I think!)

If the contour in A is outside of B, it should be kept. It seems
this case may have been missed.

PCB does not use the PBO_XOR operation, so it is difficult to test
whether this is correct or not.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 174fd17..4583732 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1593,7 +1593,7 @@ cntr_Collect (jmp_buf * e, PLINE ** A, POLYAREA ** contours, PLINE ** holes,
 	      PutContour (e, tmprev, contours, holes, owner, NULL, NULL);
 	      return TRUE;
 	    }
-	  break;
+          /* Fall through (I think this is correct. pcjc2) */
 	case PBO_UNITE:
 	case PBO_SUB:
 	  if ((*A)->Flags.status == OUTSIDE)
